apps/framework-cli/src/framework/python/wrappers/scripts/python_worker_wrapper/workflow.py

```diff
@@ -104,77 +108,78 @@ async def _execute_activity_with_state(
-                activity_name = f"{parent_dir}/{base_name}"
-                result = await self._execute_activity_with_state(
-                    activity_name, full_path, input_data
-                )
+                activity_name = f"{os.path.basename(os.path.dirname(full_path))}/{os.path.splitext(item)[0]}"
+                result = await self._execute_activity_with_state(activity_name, full_path, current_data)
                 results.append(result)
+                current_data = result  # Pass result to next step
             
             elif os.path.isdir(full_path) and "parallel" in item:
+                # Handle parallel execution
+                parallel_tasks = []
                 for script_file in sorted(os.listdir(full_path)):
                     if script_file.endswith(".py"):
-                        script_full = os.path.join(full_path, script_file)
-                        parent_dir = os.path.basename(os.path.dirname(script_full))
-                        base_name = os.path.splitext(script_file)[0]
-                        activity_name = f"{parent_dir}/{base_name}"
-                        task = self._execute_activity_with_state(
-                            activity_name, script_full, input_data
-                        )
+                        script_path = os.path.join(full_path, script_file)
+                        activity_name = f"{os.path.basename(full_path)}/{os.path.splitext(script_file)[0]}"
+                        task = self._execute_activity_with_state(activity_name, script_path, current_data)
                         parallel_tasks.append(task)
-            
-            elif os.path.isdir(full_path) and os.path.isfile(os.path.join(full_path, "child.toml")):
-                workflow_name = os.path.basename(full_path).split(".", 1)[1] if "." in os.path.basename(full_path) else os.path.basename(full_path)
-                ScriptWorkflow.__name__ = workflow_name
-                child_workflow_id = f"{workflow.info().workflow_id}:{workflow_name}"
-                
-                child_result = await workflow.execute_child_workflow(
-                    ScriptWorkflow.run,
-                    args=[os.path.join(os.path.dirname(path), workflow_name), input_data],
-                    id=child_workflow_id
-                )
                 
-                results.append(child_result)
+                if parallel_tasks:
+                    parallel_results = await asyncio.gather(*parallel_tasks)
+                    results.extend(parallel_results)
+                    # Merge parallel results if needed
+                    current_data = {
+                        "data": {k: v for result in parallel_results for k, v in result["data"].items()}
+                    }
         
-        if parallel_tasks:
-            parallel_results = await asyncio.gather(*parallel_tasks)
-            results.extend(parallel_results)
+        return results
+
+def task(retries: int = 3):
```

Comment:
Should these changes be in `tasks.py` where the decorator is defined?

Reply:
Yes, makes sense. Updated PR.